public/app/features/templating/dataMacros.ts

```diff
@@ -74,3 +119,67 @@ function getFallbackDisplayProcessor() {
 
   return fallbackDisplayProcessor;
 }
+
+interface FieldAccessorCache {
+  [key: string]: (obj: any) => any;
+}
+
+let fieldAccessorCache: FieldAccessorCache = {};
+
+function getFieldAccessor(fieldPath: string) {
+  const accessor = fieldAccessorCache[fieldPath];
+  if (accessor) {
+    return accessor;
+  }
+
+  return (fieldAccessorCache[fieldPath] = property(fieldPath));
+}
```

Comment:
Wonder if we shouldn't use single, global `fieldAccessorCache` both here and in the templateSrv, to avoid 1-1 duplication? Thinking about it, it should not cause any issues given the accessor needs to be executed against an object when needed, and my understanding of the `property` function in lodash is that it just crrates a function that will traverse given object. WDYT?

Reply:
yea, we should be able to use the same here. Can refactor that out!